pyrefly/lib/alt/solve.rs

```diff
@@ -1533,6 +1535,52 @@ impl<'a, Ans: LookupAnswer> AnswersSolver<'a, Ans> {
                     errors,
                 );
             }
+
+            // If we are inheriting from multiple base types, we should
+            // check whether the multiple inheritance is consistent
+            if class_bases.as_ref().base_type_count() > 1 {
+                // Maps field from inherited class
+                let mro = self.get_mro_for_class(cls);
+                let mut inherited_fields: SmallMap<&Name, Vec<(&Name, Type)>> = SmallMap::new();
+
+                for parent_cls in mro.ancestors_no_object().iter() {
+                    let class_fields = parent_cls.class_object().fields();
+                    for field in class_fields {
+                        let key = KeyClassField(parent_cls.class_object().index(), field.clone());
+                        let field_entry = self.get_from_class(cls, &key);
+                        if let Some(field_entry) = field_entry.as_ref() {
+                            inherited_fields
+                                .entry(field)
+                                .or_default()
+                                .push((parent_cls.name(), field_entry.ty()));
+                        }
+                    }
+                }
+
+                for (field_name, class_and_types) in inherited_fields.iter() {
+                    if class_and_types.len() > 1 {
+                        let types: Vec<Type> =
+                            class_and_types.iter().map(|(_, ty)| ty.clone()).collect();
+                        let intersect = self.intersects(&types);
+                        if matches!(intersect, Type::Never(_)) {
+                            let class_and_types_str = class_and_types
+                                .iter()
+                                .map(|(cls, ty)| {
+                                    format!("`{}` from `{}`", self.for_display(ty.clone()), cls)
+                                })
+                                .join(", ");
+                            self.error(
+                                errors,
+                                cls.range(),
+                                ErrorInfo::Kind(ErrorKind::InconsistentOverload),
+                                format!(
+                                    "Inconsistent types for field `{field_name}` inherited from multiple base classes: {class_and_types_str}",
```

Comment:
we can split this into multiple lines, 

https://github.com/facebook/pyrefly/blob/fb552b7f72d9bdced23569d8c1b3193e2f59a05f/pyrefly/lib/error/collector.rs#L121

takes a vec1 of lines